GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -491,10 +520,28 @@ List<TreeNode> SearchTree(string text, IEnumerable<TreeNode> nodes)
                     }
                 }
 
+                if (matchIndex > 0)
+                {
+                    rotateListToSetFirstAsMatchIndex(ret, matchIndex);
+                }
+
                 return ret;
             }
         }
 
+        private void rotateListToSetFirstAsMatchIndex(List<TreeNode> ret, int matchIndex)
+        {
+            int rotationIndex = 0;
+
+            while (rotationIndex < matchIndex)
+            {
+                TreeNode firstElement = ret.FirstOrDefault();
```

Comment:
Why rotate?
This changes the order in the list, this should insert the matchIndex first to keep the order

Reply:
rotation was done to maintain the tree order, in list. On click of search button multiple times, the selected node moves down  the tree, to further matches available in the list. So if I move only the match Index to first, there will be random jumping in the tree.

Reply:
At least add comments in the code to describe this

Reply:
Sure. I have added detailed comments.